src/lib/logging.ts

```diff
@@ -0,0 +1,69 @@
+import Transport from "winston-transport";
+import { createDefaultTransportFormat } from "@zwave-js/core";
+import type { ZWaveLogInfo } from "@zwave-js/core";
+import { ClientsController, Logger } from "./server";
+import { Driver } from "zwave-js";
+
+export class LoggingEventForwarder {
+  /**
+   * Only load this once the driver is ready.
+   *
+   * @param clients
+   * @param driver
+   */
+  private serverTransport?: WebSocketLogTransport;
+  public started: boolean = false;
+
+  constructor(private clients: ClientsController, private driver: Driver) {
+    // Create log transport for server
+    this.serverTransport = new WebSocketLogTransport(this.clients);
+  }
+
+  start() {
+    if (!this.serverTransport || this.serverTransport === undefined) {
+      throw new Error("Cannot start listening to logs");
+    }
+    var { transports, level } = this.driver.getLogConfig();
+    // Set the log level before attaching the transport
+    this.serverTransport.level = level as string;
+    transports = transports || [];
+    transports.push(this.serverTransport);
+    this.driver.updateLogConfig({ transports });
+    this.started = true;
+  }
+
+  stop() {
+    const transports = this.driver
+      .getLogConfig()
+      .transports.filter((transport) => transport !== this.serverTransport);
+    this.driver.updateLogConfig({ transports });
+    this.started = false;
+  }
+}
+
+class WebSocketLogTransport extends Transport {
+  private messageSymbol: Symbol;
+
+  public constructor(private clients: ClientsController) {
+    super({
+      format: createDefaultTransportFormat(false, false),
+    });
+    this.messageSymbol = Symbol.for("message");
```

Comment:
How can this symbol work. The idea of symbols is that they are unique, or do I misunderstand?

Reply:
![image](https://user-images.githubusercontent.com/7243222/115163118-51dd2f80-a075-11eb-8db1-71230c04ffd9.png)
Here's an image Al shared. He suggested I use this as a key to access the fully formatted message. I don't need to store it if that's your concern, but here's the relevant part of the docs:
`Every Symbol() call is guaranteed to return a unique Symbol. Every Symbol.for("key") call will always return the same Symbol for a given value of "key". When Symbol.for("key") is called, if a Symbol with the given key can be found in the global Symbol registry, that Symbol is returned. Otherwise, a new Symbol is created, added to the global Symbol registry under the given key, and returned.`

Reply:
TIL!